src/aruco_pose/src/aruco_pose.cpp

```diff
@@ -0,0 +1,218 @@
+/*Copyright 2024 Brookhaven National Laboratory
+BSD 3 Clause License. See LICENSE.txt for details.*/
+#include <aruco_pose/aruco_pose.hpp>
+
+using std::placeholders::_1;
+
+ArucoPose::ArucoPose()
+: Node("aruco_pose"),
+  static_broadcaster_(this), tf_broadcaster_(this),
+  LOGGER(this->get_logger())
+{
+  // Construct the camera K matrix and the distortion matrix
+  // // In OpenCV, the distortion coefficients are usually represented as a 1x8 matrix:
+  // // distCoeffs_= [k1, k2, p1, p2, k3, k4, k5, k6] where
+  // // k1, k2, k3, k3, k4, k5, k6 = radial distortion coefficients
+  // // p1, p2 = tangential distortion coefficients
+  // // For Azure kinect, they can be found when running the ROS2 camera node and
+  // // explained in the following:
+  // // https://microsoft.github.io/Azure-Kinect-Sensor-SDK/master/structk4a__calibration__intrinsic__parameters__t_1_1__param.html
+  // // These matrices are already available in the topic '/rgb/camera_info',
+  // // But it doesn't make sense to have two subscribers running to get static parameters.
+  // // Plus this avoids a situation of a different topic name being used by another vendor
+
+  // Declare parameters
+  for (const auto & param : double_params_) {
+    this->declare_parameter<double>(param, 0.0);
+  }
+  for (const auto & param : string_params_) {
+    this->declare_parameter<std::string>(param, "");
+  }
+
+  this->declare_parameter<int>("number_of_observations", 10);
+
+  cameraMatrix_ =
+    (cv::Mat_<double>(3, 3) << this->get_parameter("intrinsics.fx").as_double(), 0.0,
+    this->get_parameter("intrinsics.cx").as_double(), 0.0,
+    this->get_parameter("intrinsics.fy").as_double(),
+    this->get_parameter("intrinsics.cy").as_double(), 0, 0, 1.0);
+
+  distCoeffs_ =
+    (cv::Mat_<double>(8, 1) << this->get_parameter("dist_coeffs.k1").as_double(),
+    this->get_parameter("dist_coeffs.k2").as_double(),
+    this->get_parameter("dist_coeffs.p1").as_double(),
+    this->get_parameter("dist_coeffs.p2").as_double(),
+    this->get_parameter("dist_coeffs.k3").as_double(),
+    this->get_parameter("dist_coeffs.k4").as_double(),
+    this->get_parameter("dist_coeffs.k5").as_double(),
+    this->get_parameter("dist_coeffs.k6").as_double());
+
+  physical_marker_size_ = this->get_parameter("physical_marker_size").as_double();
+
+  double alpha = this->get_parameter("cam_rotation.alpha").as_double() / 180 * M_PI;
+  double beta = this->get_parameter("cam_rotation.beta").as_double() / 180 * M_PI;
+  double gamma = this->get_parameter("cam_rotation.gamma").as_double() / 180 * M_PI;
+
+  // Add the camera to tf server
+  camera_quaternion_.setRPY(alpha, beta, gamma);
```

Comment:
`gamma` is a member function in std

```suggestion
  double cam_alpha = this->get_parameter("cam_rotation.alpha").as_double() / 180 * M_PI;
  double cam_beta = this->get_parameter("cam_rotation.beta").as_double() / 180 * M_PI;
  double cam_gamma = this->get_parameter("cam_rotation.gamma").as_double() / 180 * M_PI;

  // Add the camera to tf server
  camera_quaternion_.setRPY(cam_alpha, cam_beta, cam_gamma);
```

Reply:
parameter names are changed:
https://github.com/maffettone/erobs/blob/5c32dc6fe6c0a002d6ecb2d3e5b09daddd49ae45/src/aruco_pose/config/fiducial_marker_param.yaml#L7-L10

https://github.com/maffettone/erobs/blob/5c32dc6fe6c0a002d6ecb2d3e5b09daddd49ae45/src/aruco_pose/src/aruco_pose.cpp#L52-L54